metrics/write.go

```diff
@@ -193,17 +196,30 @@ func updateTimetamps(tss []prompb.TimeSeries) []prompb.TimeSeries {
 	return tss
 }
 
-func collectMetrics() ([]prompb.TimeSeries, error) {
+func collectMetrics(outOfOrder bool, window time.Duration) ([]prompb.TimeSeries, error) {
 	metricsMux.Lock()
 	defer metricsMux.Unlock()
 	metricFamilies, err := promRegistry.Gather()
 	if err != nil {
 		return nil, err
 	}
-	return ToTimeSeriesSlice(metricFamilies), nil
+	tss := ToTimeSeriesSlice(metricFamilies)
+	if outOfOrder {
+		tss = shuffleTimestamps(tss, window)
+	}
+	return tss, nil
+}
+func shuffleTimestamps(tss []prompb.TimeSeries, window time.Duration) []prompb.TimeSeries {
+	now := time.Now().UnixMilli()
+	windowMillis := window.Milliseconds()
+
+	for i := range tss {
+		offset := rand.Int63n(windowMillis)
+		tss[i].Samples[0].Timestamp = now - offset
```

Comment:
This creates out of order data but its random data so its hard to then check that its what we wanted to submit.

It would be better that we consistently write back in a consistent sequence such as a few seconds and then a few minutes. For example:
- Write now() and then write 1 minute back
- write now() and then write 5 minutes back
- Repeat()